extension/src/background/messageListener/freighterApiMessageListener.ts

```diff
@@ -79,16 +80,25 @@ export const freighterApiMessageListener = (
   };
 
   const submitTransaction = async () => {
-    const { transactionXdr, network: _network, accountToSign } = request;
+    const {
+      transactionXdr,
+      network: _network,
+      networkPassphrase,
+      accountToSign,
+    } = request;
+
+    const network =
+      _network === null || !_network
+        ? MAINNET_NETWORK_DETAILS.network
+        : _network;
 
-    const network = _network ?? MAINNET_NETWORK_DETAILS.network;
     const isMainnet = getIsMainnet();
     const { networkUrl } = getNetworkDetails();
     const isExperimentalModeEnabled = getIsExperimentalModeEnabled();
     const SDK = isExperimentalModeEnabled ? SorobanSdk : StellarSdk;
     const transaction = SDK.TransactionBuilder.fromXDR(
       transactionXdr,
-      SDK.Networks[network],
+      networkPassphrase || SDK.Networks[network],
```

Comment:
allow a user to use a custom passphrase not associated with a network. Needed when a user generates a random passphrase with Stellar Quickstart